tests/unit/diff-array-test.js

```diff
@@ -0,0 +1,486 @@
+import {module, test} from 'qunit';
+
+import diffArray from 'ember-data/-private/system/diff-array';
+
+module('unit/diff-array Diff Array tests', {
+  beforeEach() {
+  },
+
+  afterEach() {
+  }
+});
+
+const a = "aaa";
+const b = "bbb";
+const c = "ccc";
+const d = "ddd";
+const e = "eee";
+const f = "fff";
+const g = "ggg";
+const h = "hhh";
+const w = "www";
+const x = "xxx";
+const y = "yyy";
+const z = "zzz";
+
+test('diff array returns no change given two empty arrays', function(assert) {
+  const result = diffArray([], []);
+  assert.equal(result.firstChangeIndex, null);
```

Comment:
I did that in my first version but thought null was more obviously "none". -1 dates from not having nullable int, thought we could be more modern.